### DISCUSSION TITLE
How to make the DropContainer refresh its items?

### DISCUSSION THREAD

**Discussion by PhilipEve** [Megabit/Blazorise#4848] [Q&A] [answered]
I haven't found a way to make the DropContainer (https://blazorise.com/docs/components/dragdrop) refresh in response to the Items property having its contents changed programmatically. The only thing that I have found to cause it to refresh is if you drag an item that is in the drop zone; the container refreshes itself after the item is dropped. But that's no good, for two reasons: (1) my user shouldn't have to do that in order to make the thing refresh; (2) the list of items is intended to start empty and have stuff added to it, so there will be nothing to drag and no way to trigger the behaviour.

Starting from the initial point where DropContainer.Items was an IReadOnlyList<TItem>, I have tried:
- Assigning DropContainer.Items to be an INotifyCollectionChanged
- Handling the CollectionChanged events myself; in the handler, creating a new list and assigning to the property that is the parameter-value for the Items parameter; calling StateHasChanged() afterwards
- Handling the CollectionChanged events myself; in the handler, creating a new Func<TItem, string, bool> and assigning to the property that is the parameter-value for the ItemsFilter parameter; calling StateHasChanged() afterwards

All I want is a list of items where the user can drag and drop items into different positions... It seems like the only way you anticipated the DropContainer being used is with a hardcoded set of items that is loaded once and then never modified, like the fruits and vegetables example on the documentation page. I'm not sure whose use case that is, but it's not one I anticipate ever having.

By the way, I can't use the forum at https://blazorise.com/support. If I try to log in, it says "Access denied! You do not have access to this resource. Try confirming your account by clicking on the link in the email we have sent you!" But you never sent me an email, and there's no "send the email again" link anywhere I could find, so there's no way to remedy the situation. I definitely entered the correct email address into the registration form, because when I try to re-register it tells me the email address is already taken. I think you should consider switching to better forums software, because the one you have at the moment seems poor.

**Comment by stsrki** [comment, marked as answer]
Did you try the `.Refresh()` on the container? First, you need to assign the reference `<DropContainer @ref="@myDropContainer">`. And then you can call the `myDropContainer.Refresh()`.

---

> By the way, I can't use the forum at https://blazorise.com/support. If I try to log in, it says "Access denied! You do not have access to this resource. Try confirming your account by clicking on the link in the email we have sent you!" But you never sent me an email, and there's no "send the email again" link anywhere I could find, so there's no way to remedy the situation. I definitely entered the correct email address into the registration form, because when I try to re-register it tells me the email address is already taken. I think you should consider switching to better forums software, because the one you have at the moment seems poor.

The forum is only meant for our commercial users at the moment. I have sent you a new verification email so please look into your inbox, including the spam folder.

As for changing the software that is not possible because it is our custom solution with an integrated system for our own company and teams.

**Reply by PhilipEve** [reply]
No, I did not try .Refresh(). It works. It looks very obvious in retrospect, but in my defense, that method is mentioned nowhere on the documentation page at https://blazorise.com/docs/components/dragdrop. The word "refresh" doesn't appear once on the page if I ctrl+F for it!

Thank you for the pointer.

**Reply by stsrki** [reply]
Thanks for letting us know that we need to update our documentation :)
